class.jetpack-gutenberg.php

```diff
@@ -58,12 +60,76 @@ public static function load_blocks() {
 			return;
 		}
 
-		foreach ( self::$blocks as $type => $args ) {
-			register_block_type(
-				'jetpack/' . $type,
-				$args
+		/**
+		 * Filter the list of blocks that are available though jetpack.
+		 *
+		 * This filter is populated by Jetpack_Gutenberg::jetpack_set_available_blocks
+		 *
+		 * @since 6.8.0
+		 *
+		 * @param array
+		 */
+		self::$blocks_index = apply_filters( 'jetpack_set_available_blocks', array() );
+
+		foreach ( self::$jetpack_blocks as $type => $args ) {
+			if ( isset( $args['availability']['available'] ) && $args['availability']['available'] && in_array( $type, self::$blocks_index ) ) {
+				register_block_type( 'jetpack/' . $type, $args['args'] );
+			}
+		}
+	}
+
+	public static function preset_exists( $preset ) {
```

Comment:
Could you add docblocks for all those new functions? Lots of people will be looking up to Jetpack and checking how we did things when we start releasing our blocks, so it would be nice to have something that looks good (no linter warnings in IDE) and is as self-explanatory as possible. It will also help address #10540 imo.